packages/theme/src/types.ts

```diff
@@ -87,3 +87,24 @@ export type SimpleThemeOptions = {
   palette: BackstagePaletteOptions;
   fontFamily?: string;
 };
+
+export type PageTheme = {
```

Comment:
We somehow want to be providing this type to consumers, but still allow it to be configured in the App. One possibility would be to merge it into the main theme, essentially having `theme.pageTheme` on the `BackstageTheme`. The theme itself would define the default page theme, and then the individual pages would wrap themselves in a MUI `ThemeProvider` to override to the custom theme for each page.

At that point the problem becomes how you select the theme for each page, and open to suggestions there :grin:
One way I could think of is to simply make a `EntityTheme` API, which implements `(entity: Entity) => PageTheme`, but that may be a bit too limiting.

Reply:
> but still allow it to be configured in the App.

Can you please elaborate this sentence. I am not getting it. I mean there is already configuration in theme package, don't they have flexiblity to modify from there?

Reply:
Same way as theming works in MUI, https://material-ui.com/customization/theming/#theme-provider, which ofc is what we're actually doing in `@backstage/theme`. The key part is that we're providing and consuming themes through React contexts, not by direct imports. By doing it through context you allow for components higher up in the tree to override the theme, which in our case is done at the root level in the app. [this](https://github.com/spotify/backstage/blob/ac22f385b8e50d579a74521c3ad4ee880a446c9a/packages/core-api/src/app/AppThemeProvider.tsx#L93) is where `@backstage/core` provides the theme for the app.